src/BenchmarkDotNet/Reports/SummaryTableExtensions.cs

```diff
@@ -57,6 +57,9 @@ public static void PrintLine(this SummaryTable table, string[] line, ILogger log
                     continue;
                 }
 
+                if (escapePipe)
+                    line[columnIndex] = line[columnIndex].Replace("|", "\\|");
```

Comment:
by replacing the original value with the new one the escaping is applied to all other exporters who later use the same data